case-study.Rmd

```diff
@@ -0,0 +1,489 @@
+                                               model = model)
+DT::datatable(sig_annotation)
+```
+
+#### Associate pathways
+Next we can associate existing pathways to signatures. Here we use KEGG pathways.
+We retrive *Pseudomonas aeruginosa* KEGG terms from the
+[TRIBE](http://tribe.greenelab.com/#/home) webserver. TRIBE periodically updates
+KEGG terms. To reproduce results presented in the paper, we use the KEGG
+pathways that were retrieved from TRIBE on June 03 2016.
+You can also repeat the analysis with up-to-date KEGG terms from TRIBE.
+TRIBE also supports Gene Ontology terms. You can simply replace "KEGG" with "GO"
+in the following steps to repeat the analysis with GO terms.
+
+```{r}
+KEGG <- fetch_geneset(type = "KEGG", access_date = "06-03-16")
+# we only consider KEGG pathways with more than 5 genes and less than 100 genes
+# as meaningful pathways
+KEGG_subset <- KEGG[lengths(KEGG) >= 5 & lengths(KEGG) <= 100]
+```
+
+We associate active signatures to known KEGG pathways.
+
+```{r}
+pathway_association <- annotate_signatures_with_genesets(
+  selected_signatures = unique_active_sigs, model = model,
+  genesets = KEGG_subset)
+```
+
+Calculate the activity of associated pathways inside active signatures. This help
+differentiate active vs. non-active pathways assoicated with active signatures.
+
+```{r, fig.height= 10, fig.width=10}
+pathway_activity <- signature_geneset_activity(
+  signature_geneset_df = pathway_association[, c("signature", "geneset")],
+  gene_set_list = KEGG_subset, model = model, input_data = data_normed)
+plot_activity_heatmap(pathway_activity)
+```
+
+We run a limma test on pathway activities and find pathways that are truly
+active in this dataset.
+
+```{r}
+pathway_limma <- build_limma(pathway_activity, phenotypes = data_pheno,
+                             control_pheno = "wt", use.bonferroni = TRUE)
+# combine pathway association and pathway activation test results
+combined_result <- combine_geneset_outputs(
+  signature_geneset_association = pathway_association,
+  geneset_limma_result = pathway_limma)
+knitr::kable(combined_result, digits = 4, row.names = FALSE, align = "c")
+```
+
+#### Gene-gene networks
+We next visualize how genes in the active signatures cluster in the
+ADAGE-derived gene-gene network.
+
+We calculate an expression fold change for each gene and pass it to the
+gene-gene network to show as node color. Again, we use limma to test
+differential expression and get the logFC.
+
+```{r}
+data_raw_limma <- build_limma(input_data = data_raw, phenotypes = data_pheno,
+                              control_pheno = "wt")
+# build a gene:fold change table from limma result
+gene_logFC <- data.frame(geneID = rownames(data_raw_limma),
+                         logFC = data_raw_limma$logFC)
+```
+
+Visualize the ADAGE gene-gene network of the active signatures.
+
+```{r}
+visualize_gene_network(selected_signatures = unique_active_sigs,
```

Comment:
It's a good suggestion but I don't see how to do it using the current network visualization library.